test/provider-kms.js

```diff
@@ -83,12 +84,25 @@ describe('Provider/KMS', function() {
       provider._client.config.region.should.equal(region);
       global.Config = _Config;
     });
+
+    it('sets the _datakey property if a datakey value is provided', function() {
+      const provider = new KMSProvider({ciphertext: 'foo', region: 'us-east-1', datakey: 'foobar'});
+
+      provider._datakey.should.be.true();
+    });
+
+    it('sets the CiphertextBlob property of _parameters to the base64 encoded datakey', function() {
+      const provider = new KMSProvider({ciphertext: 'foo', region: 'us-east-1', datakey: 'foobar'});
+
+      provider._parameters.CiphertextBlob.should.eql(Buffer.from('foobar', 'base64'));
+    });
   });
 
   describe('KMSProvider#initialize', function() {
+    const plaintext = Buffer.from('this-is-a-secret', 'utf-8').toString('base64');
     const validResponse = {
       KeyId: 'arn:aws:kms:us-east-1:ACCOUNT:key/SOME-UUID',
-      PlainText: Buffer.from('this-is-a-secret', 'utf8').toString('base64')
+      Plaintext: Buffer.from(plaintext, 'base64')
```

Comment:
Turns out I mucked it up in the original tests. The value returned from [`KMS.decrypt()`](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/KMS.html#decrypt-property) is actually `Plaintext`.